sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1077,6 +1077,27 @@ object SparkSession extends Logging {
       throw new IllegalStateException("No active or default Spark session found")))
   }
 
+  /**
+   * Returns a cloned SparkSession with all specified configurations disabled, or
+   * the original SparkSession if all configurations are already disabled.
+   *
+   * @since 3.1.0
+   */
+  def getOrCloneSessionWithConfigsOff(
```

Comment:
`private[spark]`.

Reply:
curious why we need to add this? what's the issue we are preventing? also why `private[spark]` but not `private[sql]`?

Reply:
Oh, `private[sql]` is okay too. I don't think we should expose this as public.

Reply:
@viirya - sure, updated.